src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBase.java

```diff
@@ -313,36 +309,13 @@ public String getAuthorizedPublicKey() {
                 "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDXRix6tfUq1K3IbgXYMqKYdgwYU8hxgUiBwDtBpxq7smPT5s5xCg7C6vdYIitDfgLORbWheyCuHgOErwyoh2GAm842di5BjJVsw+WgxLUcM0gkUM06PJdM7I9hBuz8sxZM7Kcose382QSNWwg5cTQFhLzZr8ig5OtSYc6PPAjzR4GiX93akoZLcFaMmXZba8Qdqhr+zsrmCtgfxqAIJTXzVxtz0MLscTDk37ZtnRVnKXOOvy/a7hZumgKvTom+mSv0kq6hVNlUv2Q9jmNHp/RfLU5xc3aN0BninRXx7yaTfqXwhCZHfgbhG+B8PpMWZvwUBDsakxsVyGdEeGQo+n03 testkey-no-passphrase";
     }
 
-    /**
-     * Returns a valid port number to use in Mocked SSH Server. Verifies that the port is not being used.
-     *
-     * @return int with a valid port number
-     */
-    private int getValidPort() throws IOException {
```

Comment:
the port may be free immediately after it was obtained - but may have been used by something else before it was used by the SSH server.